sklearn/tree/_export.py

```diff
@@ -986,7 +991,10 @@ def export_text(
     check_is_fitted(decision_tree)
     tree_ = decision_tree.tree_
     if is_classifier(decision_tree):
-        class_names = decision_tree.classes_
+        if class_names is not None and len(class_names) == len(decision_tree.classes_):
+            class_names = class_names
+        else:
```

Comment:
This else catches the case where a user provides the class_names arg, but `len(class_names) == len(decision_tree.classes_)` - in that case, would it be worth raising an error to alert the user of this? Otherwise it will not be clear why, despite the fact that they passed class_names, they are not being used as the user would expect. What do you think?

Reply:
Indeed, it can be interesting, this would allow the user to insert a list with the correct number of arguments. Otherwise, it would not have the desired effect.

```python
    if is_classifier(decision_tree):
        if class_names is not None:
            if len(class_names) != len(decision_tree.classes_):
                raise ValueError(
                    "class_names must contain %d elements, got %d"
                    % (tree_.n_classes, len(class_names))
                )
        else:
            class_names = decision_tree.classes_
```

So here, if `class_names` is not None and there is the right number of arguments (i.e. equal to the number of classes) then class_names can continue the operation of the function. What do you think ?

Reply:
Yes exactly, something like the above looks sensible to me. You could combine the two if statements on lines 2 and 3 into one line too.